### PR TITLE
docs(button): example toggle button practice

### PR DESCRIPTION
**Pull request by yinonov**

### CODE REVIEW

**Review by rinaok — COMMENTED** [review]
Why is the toggle button under usage examples and not components?

**Review by rinaok — CHANGES_REQUESTED** [review]
.

**Review by rinaok — APPROVED** [review]

**Comment by rachelbt** [comment]
What about the "on" state? 
image
image
image

In only text button - if the text doesn't change - how can a user know that the button is toggled? Are we leaving it to our consumers to decide and switch between appearances?

**Comment by yinonov** [comment]
> What about the "on" state? image image image
> 
> In only text button - if the text doesn't change - how can a user know that the button is toggled? Are we leaving it to our consumers to decide and switch between appearances?

yes, it's for consumers to decide. the point here is to reflect how easy it is to tweak a button to be toggle-able.

1 of the issues I experience here is that the _toggled-to_ icon isn't preloaded. thus, firstly shows the preloading spinner

**Comment by YonatanKra** [comment]
> > What about the "on" state? image image image
> > In only text button - if the text doesn't change - how can a user know that the button is toggled? Are we leaving it to our consumers to decide and switch between appearances?
> 
> yes, it's for consumers to decide. the point here is to reflect how easy it is to tweak a button to be toggle-able.
> 
> 1 of the issues I experience here is that the _toggled-to_ icon isn't preloaded. thus, firstly shows the preloading spinner

Won't it always be a problem? I mean, we can't control how fast a user downloads an icon...

File: docs/usage-examples/toggle-button.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  id="button"
+  connotation='cta'
+  shape='pill'
+  label='not pressed'
+  aria-label="microphone button">
+</vwc-button>
+
+<script>
+  const toggleWithLabel = ({target}) => {
+    target.ariaPressed = !target.ariaPressed;
```
Comment (YonatanKra) [inline docs/usage-examples/toggle-button.md:16]: This is actually one of the values of `vivid` - a developer would hardly think of changing the aria pressed value...

Comment (yinonov) [reply docs/usage-examples/toggle-button.md:16]: ure right, but does this tiny "program" justify a dedicated component?

**Comment by yinonov** [comment]
> > > What about the "on" state? image image image
> > > In only text button - if the text doesn't change - how can a user know that the button is toggled? Are we leaving it to our consumers to decide and switch between appearances?
> > 
> > 
> > yes, it's for consumers to decide. the point here is to reflect how easy it is to tweak a button to be toggle-able.
> > 1 of the issues I experience here is that the _toggled-to_ icon isn't preloaded. thus, firstly shows the preloading spinner
> 
> Won't it always be a problem? I mean, we can't control how fast a user downloads an icon...

I think this might introduce a lazy/eager loading mechanism for icon component. maybe a task of its own?

**Review by YonatanKra — APPROVED** [review]

**Pull request merged**